src/user/user.repository.ts

```diff
@@ -0,0 +1,93 @@
+import { buildConnection } from '../database';
+import { User } from './user.type';
+
+export const findByLogin = async (login: string): Promise<User | null> => {
+  const dbConnection = buildConnection();
+
+  const user = await dbConnection.oneOrNone(
+    `
+      SELECT id FROM users
+      WHERE login LIKE $1
+  `,
+    login
+  );
+
+  return user;
+};
+
+export const insert = async (user: User): Promise<User> => {
+  const { login, name, location, url, createdAt } = user;
+  const dbConnection = buildConnection();
+
+  return dbConnection.one(
+    `
+      INSERT INTO users(login, name, location, url, created_at) 
+      VALUES ($1, $2, $3, $4, $5)
+      RETURNING id
+    `,
+    [login, name, location, url, createdAt]
+  );
+};
+
+export const insertManyUserLanguages = async (
+  userId: number,
+  languages: string[]
+): Promise<void> => {
```

Comment:
Why are you creating a database connection on each operation, instead of recycling an already existing connection?
What are the risks of your current approach?

Reply:
I'm not creating it every time. I've saved it in a variable on `database/index.ts`. Honestly, I didn't find a better way of doing this while having env variables and a functional approach.
Maybe the naming could be a bit more clearer, instead of buildConnection, it could be getConnection.